presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8153,4 +8165,49 @@ public void testAssignUniqueId()
                         "WHERE a = 1)",
                 "VALUES 3008750");
     }
+
+    @Test
+    public void testAggregationPushedBelowOuterJoin()
+    {
+        assertQuery(
+                "SELECT * " +
+                        "FROM nation n1 " +
+                        "WHERE (n1.nationkey > ( " +
+                        "SELECT avg(nationkey) " +
+                        "FROM nation n2 " +
+                        "WHERE n1.regionkey=n2.regionkey))"
+        );
+        assertQuery(
+                "SELECT max(nationkey) " +
+                        "FROM (SELECT DISTINCT regionkey FROM region) as r1 " +
+                        "LEFT JOIN " +
+                        "nation " +
+                        "ON r1.regionkey = nation.regionkey " +
+                        "GROUP BY r1.regionkey");
+
+        assertQuery(
+                "SELECT max(nationkey) " +
+                        "FROM (SELECT regionkey FROM region INTERSECT SELECT regionkey FROM region where regionkey < 4) as r1 " +
+                        "LEFT JOIN " +
+                        "nation " +
+                        "ON r1.regionkey = nation.regionkey " +
+                        "GROUP BY r1.regionkey");
+
+        assertQuery(
+                "SELECT max(nationkey) " +
+                        "FROM (SELECT regionkey FROM region EXCEPT SELECT regionkey FROM region where regionkey < 4) as r1 " +
+                        "LEFT JOIN " +
+                        "nation " +
+                        "ON r1.regionkey = nation.regionkey " +
+                        "GROUP BY r1.regionkey");
+
+        assertQuery(
+                "SELECT max(nationkey) " +
+                        "FROM (VALUES 1) v1(col1) " +
+                        "LEFT JOIN " +
+                        "nation " +
+                        "ON v1.col1 = nation.regionkey " +
+                        "GROUP BY v1.col1",
+                "VALUES 24");
+    }
```

Comment:
done.  Didn't add a test for filtered aggregation because once you add a filter the outer join gets converted to an inner join so it doesn't go through this optimizer.
Though,  I guess this can be extended to inner joins too if you group on all columns from one of the tables, etc. (and wouldn't need the cross join bit).  I'll think it through and maybe do as a follow on.